### ISSUE TITLE
New Rule: single-h1

### ISSUE THREAD

**Issue by Pixel998** [eslint/markdown#373]
### Rule details

This rule enforces that a Markdown document contains at most one top-level heading.

### What type of rule is this?

Warns about a potential problem

### Example code

```markdown
Examples of incorrect code:

# Top level heading

# Another top-level heading
```

### Participation

- [x] I am willing to submit a pull request to implement this rule.

### Additional comments

Rationale: Using multiple h1 elements can disrupt the semantic hierarchy of a document. Search engines and assistive technologies rely on a clear heading structure to understand the content's organization. A single h1 clearly identifies the primary topic.

Prior Art:
[MD025](https://github.com/DavidAnson/markdownlint/blob/main/doc/md025.md)

**Comment by nzakas** [comment]
Sounds good to me. 👍
